3.Data-classes/Data_Classes

```diff
@@ -0,0 +1,176 @@
+
+
+#1. Create a method which will take a natural number as an argument and will find a sum of all its digits
+  def sum_of_numbers(natural_number="93846675")
+    array = natural_number.each_char.to_a
+    index = 0
+    sum = 0
+    while index <= array.count
+      sum = sum + array[index].to_i
+      index = index + 1
+    end
+    puts sum
+  end
+
+puts "method #1"
+sum_of_numbers
+sum_of_numbers("111")
+
+
+
+
+
+#2. There is a string. Find a maximum number of digits placed in a row in it
+  def find_max_number_of_digits(string="00 000 0")
+    array = string.gsub(/[^\d]/, '|').split('|')
+    max_number_of_digits = Array.new
+    array.each {|el| max_number_of_digits.push(el.length) }
+    puts max_number_of_digits.max
+  end
+
+puts "method #2"
+find_max_number_of_digits
+find_max_number_of_digits("lalala oo jket 99hejjjruuc")
+
+
+
+
+
+#3. There is a string. Find a maximal number (not a digit) in it
+  def max_number_in_string(string="89563 4777 10")
+    array = string.gsub(/[^\d]/, '|').split('|')
+    new_array = Array.new
+    array.each {|el| new_array.push(el.to_i) }
+    puts new_array.max
+  end
+
+puts "method #3"
+max_number_in_string
+max_number_in_string("8956679 76544 88")
+
+
+
+
+
+#4. There are two strings. Find number of first chars from first string matching first chars of the second string. Consider two cases:
+#strings are definitely different
+#strings can completely match
+def find_number_of_first_chars(string1="First string", string2="Second string")
+  arr1 = string1.split('')
+  arr2 = string2.split('')
+  counter = 0
+  arr1.each_with_index do |el, i|
+    if arr1[i] == arr2[i]
+      counter +=1
+      else next
+    end
+  end
+  print counter
+end
+
+puts "method #4"
+puts find_number_of_first_chars
+puts find_number_of_first_chars("Wow, cucumber", "Wow, cucumber")
+
+
+
+
+
+#5. There is an array of integers. First puts elements with even indexes and then with odd indexes
+  def sort_indexes(array_of_integers="123456789")
+    arr = array_of_integers.each_char.to_a
+    even_arr = Array.new
+    odd_arr = Array.new
+      for i in 0..(arr.length - 1) do
```

Comment:
'for' loop is not recommended to use in ruby because of 'i' variable is out of 'do...end' block (ask personally about details)
ruby has a plenty of cool loop expressions: each, loop, while, until, n.times, etc
so please find a most suitable for you and use it here